File: app/back/src/main/java/net/seijishikin/jp/normalize/manage/kanrensha/dto/postal/SearchWkTblPostalCodeCapsuleDto.java
package net.seijishikin.jp.normalize.manage.kanrensha.dto.postal;

import java.io.Serializable;

import net.seijishikin.jp.normalize.common_tool.dto.FrameworkCapsuleDto;
import net.seijishikin.jp.normalize.common_tool.dto.paging.PagingIntegerDtoInterface;

/**
 * 郵便番号差分ワークテーブル検索条件Dto
 */
public class SearchWkTblPostalCodeCapsuleDto extends FrameworkCapsuleDto // NOPMD DataClass
        implements Serializable, PagingIntegerDtoInterface {

    /** Serialize id */
    private static final long serialVersionUID = 1L;

    /** 全件数 */
    private Integer allCount = INIT_INTEGER;

    /** 抽出件数 */
    private Integer limit = INIT_INTEGER;

    /** ページ番号 */
    private Integer pageNumber = INIT_INTEGER;

    /**
     * 全件数を取得する
     *
     * @return 全件数
     */
    @Override
    public Integer getAllCount() {
        return allCount;
    }

    /**
     * 全件数を設定する
     *
     * @param allCount 全件数全件数
     */
    @Override
    public void setAllCount(final Integer allCount) {
        this.allCount = allCount;
    }

    /**
     * 抽出件数を取得する
     *
     * @return 抽出件数
     */
    @Override
    public Integer getLimit() {
        return limit;
    }

    /**
     * 抽出件数を設定する
     *
     * @param limit 抽出件数
     */
    @Override
    public void setLimit(final Integer limit) {
        this.limit = limit;
    }

    /**
     * ページ番号を取得する
     *
     * @return ページ番号
     */
    @Override
    public Integer getPageNumber() {
        return pageNumber;
    }

    /**
     * ページ番号を設定する
     *
     * @param pageNumber ページ番号
     */
    @Override
    public void setPageNumber(final Integer pageNumber) {
        this.pageNumber = pageNumber;
    }

    /** 履歴検索該否 */
    private Boolean isSearchHistory = INIT_BOOLEAN;

    /**
     * 履歴検索該否
     * 
     * @return 履歴検索該否
     */
    public Boolean getIsSearchHistory() {
        return isSearchHistory;
    }

    /**
     * 履歴検索該否
     * 
     * @param isSearchHistory 履歴検索該否
     */
    public void setIsSearchHistory(final Boolean isSearchHistory) {
        this.isSearchHistory = isSearchHistory;
    }


    /** 自動修復検索該否 */
    private Boolean isSearchRepair = INIT_BOOLEAN;

    /**
     * 自動修復検索該否を取得する
     * 
     * @return 自動修復検索該否
     */
    public Boolean getIsSearchRepair() {
        return isSearchRepair;
    }

    /**
     * 自動修復検索該否を設定する
     * 
     * @param isSearchRepair 自動修復検索該否
     */
    public void setIsSearchRepair(final Boolean isSearchRepair) {
        this.isSearchRepair = isSearchRepair;
    }

}

File: app/front/src/components/main/dto/address_postal/searchWkTblPostalCodeCapsuleDto.ts
import { FrameworkCapsuleDto, type FrameworkCapsuleDtoInterface, type FrameworkPagingDtoInterface } from "seijishikin-jp-normalize_common-tool";

interface SearchWkTblPostalCodeCapsuleDtoInterface
    extends FrameworkPagingDtoInterface, FrameworkCapsuleDtoInterface {

    /** 履歴検索該否 */
    isSearchHistory: boolean;

    /** 自動修復検索該否 */
    isSearchRepair: boolean;

    /** 原文書名 */
    orgName: string;

    /** 編集内容説明 */
    worksText: string;
}

class SearchWkTblPostalCodeCapsuleDto extends FrameworkCapsuleDto
    implements SearchWkTblPostalCodeCapsuleDtoInterface {

    allCount: number;
    limit: number;
    pageNumber: number;

    /** 履歴検索該否 */
    isSearchHistory: boolean;

    /** 自動修復検索該否 */
    isSearchRepair: boolean;

    /** 原文書名 */
    orgName: string;

    /** 編集内容説明 */
    worksText: string;

    constructor() {
        super();
        const INIT_NUMBER: number = 0;
        const INIT_STRING: string = "";
        const INIT_BOOLEAN: boolean = false;

        this.isSearchHistory = INIT_BOOLEAN;
        this.isSearchRepair = INIT_BOOLEAN;
        this.allCount = INIT_NUMBER;
        this.limit = INIT_NUMBER;
        this.pageNumber = INIT_NUMBER;
        this.orgName = INIT_STRING;
        this.worksText = INIT_STRING;
    }
}

export { type SearchWkTblPostalCodeCapsuleDtoInterface, SearchWkTblPostalCodeCapsuleDto }
